### PR TITLE
[Semantics] Allow metadata to be queried by type, with defaults

### PR DESCRIPTION
**Pull request by c-harding**
The old semantic parsed output was not easy to query, so now querying functionality has been added, in the form of `serviceBlock.lookupMetadata[Language]`.

### CODE REVIEW

File: src/main/scala/temple/DSL/semantics/TempleBlock.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    metadata.iterator.collectFirst { case m: T => m }
+
+  /**
+    * Find a metadata item by type, defaulting to the value defined at the project level
+    * @tparam T The type of metadata to be provided. This must be explicitly given, in square brackets
+    * @return an option of the metadata item
+    */
+  def lookupMetadata[T <: Metadata: ClassTag]: Option[T] =
+    lookupLocalMetadata[T] orElse lookupDefaultMetadata[T]
+}
```
Comment (jaylees14) [inline src/main/scala/temple/DSL/semantics/TempleBlock.scala:42]: Could we have some tests for this?

**Review by jaylees14 — APPROVED** [review]

File: src/main/scala/temple/DSL/semantics/TempleBlock.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Option.when(parent.projectBlock != this)(parent) flatMap { _.projectBlock.lookupMetadata[T] }
+  }
+
+  /**
+    * Find a metadata item by type
+    * @tparam T The type of metadata to be provided. This must be explicitly given, in square brackets
+    * @return an option of the metadata item
+    */
+  final def lookupLocalMetadata[T <: Metadata: ClassTag]: Option[T] =
+    metadata.iterator.collectFirst { case m: T => m }
```
Comment (Lewiky) [inline src/main/scala/temple/DSL/semantics/TempleBlock.scala:33]: This is smart

**Review by Lewiky — APPROVED** [review]
LGTM, Some tests would be good like Jay said

**Pull request merged**